libvips/foreign/magicksave.c

```diff
@@ -465,8 +465,15 @@ vips_foreign_save_magick_buffer_build( VipsObject *object )
 		build( object ) )
 		return( -1 );
 
-	if( !(obuf = ImagesToBlob( magick->image_info, magick->images, 
-		&olen, magick->exception )) ) { 
+#ifdef HAVE_IMAGESTOBLOB
+	obuf = ImagesToBlob( magick->image_info, magick->images,
+		&olen, magick->exception );
+#else
+	obuf = ImageToBlob( magick->image_info, magick->images,
+		&olen, magick->exception );
+#endif /*HAVE_IMAGESTOBLOB*/
```

Comment:
I think it'd be best to make a little wrapper in magick.c called `magick_images_to_blob()` and call that here. 

The aim of magick.c is to get all these ugly ifdefs out of the main body of the code.